rotkehlchen/tasks/events.py

```diff
@@ -16,14 +35,146 @@ def process_events(
 
     This is supposed to be a generic processing task that can be requested or run periodically
     """
-    eth2 = chains_aggregator.get_module('eth2')
-    if eth2 is not None:
+    if (eth2 := chains_aggregator.get_module('eth2')) is not None:
         eth2.combine_block_with_tx_events()
         eth2.refresh_activated_validators_deposits()
 
+    match_asset_movements(database)
+
     with database.user_write() as write_cursor:
-        database.set_static_cache(  # update last withdrawal query timestamp
+        database.set_static_cache(  # update last event processing timestamp
             write_cursor=write_cursor,
             name=DBCacheStatic.LAST_EVENTS_PROCESSING_TASK_TS,
             value=ts_now(),
         )
+
+
+def match_asset_movements(database: 'DBHandler') -> None:
+    """Analyze asset movements and find corresponding onchain events, then update those onchain
+    events with proper event_type, counterparty, etc and cache the matched identifiers.
+    """
+    log.debug('Analyzing asset movements for corresponding onchain events...')
+    events_db = DBHistoryEvents(database=database)
+
+    # get all exchange asset movements
+    asset_movements: list[AssetMovement]
+    with database.conn.read_ctx() as cursor:
+        if len(asset_movements := events_db.get_history_events_internal(  # type: ignore  # will be asset movements due to entry type filter.
+            cursor=cursor,
+            filter_query=HistoryEventFilterQuery.make(entry_types=IncludeExcludeFilterData(
+                values=[HistoryBaseEntryType.ASSET_MOVEMENT_EVENT],
```

Comment:
couple questions:
1. shouldn't fee events be excluded from the result here?
2. if fees are excluded, how would you properly match a deposit to an onchain event? there'd be a difference equal to the fee amount, and i don't think that's currently accounted for.